### PR TITLE
[language] Implement an API to fetch external dependencies of a Trans…

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Currently the Move IR to bytecode compiler isn't connected to the on-chain storage, thus it can't resolve the on-chain dependencies of a module. This diff will be the very first step to support such feature for a friendly dev experience. The plan is that after this diff, we will use CLI to fetch the modules on chain with this API, and pass them back to the compiler.

NOTE: The current api will get ALL the on-chain dependencies, including those in the stdlib. Thus we might want to use no-stdlib flag when compiling the script with the fetched modules. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
We will test it once we have the publish&execute pipeline working.

### CODE REVIEW

File: language/compiler/ir_to_bytecode/syntax/src/ast.rs [hunk 1]
```diff
@@ -523,6 +526,23 @@ impl Script {
             FunctionBody::Native => panic!("main() can't be native"),
         }
     }
+
+    /// Return a vector of `AccessPath` where the on chain dependent modules are stored.
+    pub fn get_on_chain_deps(&self) -> Vec<AccessPath> {
```
Comment (sblackshear) [inline language/compiler/ir_to_bytecode/syntax/src/ast.rs]: Would it make more sense for this to return a vector of `ModuleId`'s? That way, we could implement the same function for `CompiledModule` and share the `ModuleId` -> `AccessPath` logic in both.

Also: perhaps name something like `get_module_dependencies`? The `on_chain` part is a bit confusing to me given that this doesn't actually read from the blockchain...

File: language/compiler/ir_to_bytecode/syntax/src/ast.rs [hunk 2]
```diff
@@ -523,6 +526,23 @@ impl Script {
...
+        let mut deps = vec![];
```
Comment (sblackshear) [inline language/compiler/ir_to_bytecode/syntax/src/ast.rs]: Do we care if there are duplicate imports? We might want to return a set or deduplicate the return vector.

File: language/compiler/ir_to_bytecode/syntax/src/ast.rs [hunk 3]
```diff
@@ -523,6 +526,23 @@ impl Script {
...
+        for dep in self.imports.iter() {
+            match &dep.ident {
+                ModuleIdent::Qualified(id) => {
+                    deps.push(AccessPath::code_access_path(&ModuleId::new(
+                        id.address,
+                        id.name.name(),
+                    )));
+                }
+                _ => (),
```
Comment (sblackshear) [inline language/compiler/ir_to_bytecode/syntax/src/ast.rs]: Nit: explicit pattern matching is safe. It lets the compiler tell us when we add variants that need to be handled here in the future.

**Comment by bmwill** [comment]
Please see the [contributing doc](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#responding-to-reviewer-feedback) for how to respond to reviewer feedback as you shouldn't be making separate commits on top to satisfy ci

File: language/compiler/ir_to_bytecode/syntax/src/ast.rs [hunk 4]
```diff
@@ -503,6 +503,16 @@ pub type Exp_ = Spanned<Exp>;
 // impls
 //**************************************************************************************************
 
+fn get_external_deps(imports: &[ImportDefinition]) -> Vec<ModuleId> {
```
Comment (sblackshear) [inline language/compiler/ir_to_bytecode/syntax/src/ast.rs]: _y_

**Pull request merged**